### PR TITLE
changed ft_strncmp to ft_strcmp in is_builtin function. modified buil…

### PR DESCRIPTION
**Pull request by shoko-soeno**
…tin_exit to store exit codes similar to bash
#4 

## Release Notes

- **New Features**
  - Enhanced `exit` command with improved argument validation.
  - Added comprehensive checks for exit argument types and formats.
  - Introduced a new function to reset the command context.

- **Bug Fixes**
  - Improved error handling for exit command scenarios.
  - Enhanced formatting of error messages for better readability.

- **Tests**
  - Added extensive test cases for `exit` command covering various input scenarios.
  - Expanded test coverage for edge case handling.

- **Documentation**
  - Updated file headers and timestamps across multiple source files.

### CODE REVIEW

File: srcs/builtin/builtin_exit.c [hunk 1]
```diff
@@ -6,21 +6,62 @@
...
+	if (argc == 2)
+	{
+		if (!is_digit_str(argv[1]) || argv[1] == NULL)
+		{
+			builtin_error("exit", argv[1], "numeric argument required");
+			ctx->last_status = EXIT_INVALID_INPUT;
+			return (ctx->last_status);
+		}
+		ctx->last_status = ft_atoi(argv[1]);
+	}
```
[the remark this answers is not in the document]
Comment (shoko-soeno) [reply srcs/builtin/builtin_exit.c]: fixed
